src/XmlReader.php

```diff
@@ -265,15 +269,27 @@ public function element(string $name, array $withAttributes = []): LazyQuery
     /**
      * Search for an element with xpath
      *
+     * @param array<string, string> $namespaceMap
      * @throws \Throwable
      * @throws \VeeWee\Xml\Encoding\Exception\EncodingException
      */
-    public function xpathElement(string $query): Query
+    public function xpathElement(string $query, array $namespaceMap = []): Query
     {
         try {
             $xml = iterator_to_array($this->reader->provide(Matcher\document_element()))[0];
 
-            $xpath = Document::fromXmlString($xml)->xpath();
+            $configurators = [];
+
+            // When the namespace map is empty we will remove the root namespaces
+            // because if they are not mapped then you cannot search on them.
+
+            if (empty($namespaceMap)) {
+                $xml = xml_encode(xml_decode($xml, traverse(RemoveNamespaces::unprefixed())));
```

Comment:
@veewee is this okay? I saw that I could use `$document->traverse()` but I couldn't quite get it working. Ideally I'd like to run the traverse and then pass into the XPath method

Reply:
<img width="411" alt="Screenshot 2023-11-05 at 14 45 27" src="https://github.com/saloonphp/xml-wrangler/assets/29132017/6f9019ec-7bb8-4453-8356-83bbed1736f4">
Can I do something like this?

Reply:
It would rather look like this:

```php
$xmlConfigurators = [];
$xpathConfigurators = [];


// When the namespace map is empty we will remove the root namespaces
// because if they are not mapped then you cannot search on them.

if (empty($namespaceMap)) {
    $xmlConfigurators[] = traverse(RemoveNamespaces::unprefixed());
} else {
    $xpathConfigurators[] = namespaces($namespaceMap);
}

$xpath = Document::fromXmlString($xml, ...$xmlConfigurators)->xpath(...$xpathConfigurators);
```

There is no need to decode and encode it back again : the configurators work at DOM level (as you can see in the namespace) - since the encoding component is using the DOM component internally, it just happens to all work together :)

Reply:
I think I've figured it out. When I encode/decode and remove the namespaces before passing into the DOM, the service node will just be called "service". However with the newer, better way - it seems to prefix it with "default". Maybe it's something to do with the order of processing? 

Old:

<img width="630" alt="Screenshot 2023-11-05 at 15 14 29" src="https://github.com/saloonphp/xml-wrangler/assets/29132017/7c04f6ed-4881-4959-a00a-553e6df70409">

New: (Not working)

<img width="466" alt="Screenshot 2023-11-05 at 15 16 10" src="https://github.com/saloonphp/xml-wrangler/assets/29132017/2713d366-342e-474a-bae5-ccd2587cd822">

Reply:
Don't know the reason for that. I've seen something similar before, but will have to dive into the specifics again. Maybe later this week

Reply:
Going to release it with the `xml_decode(xml_encode(...))` to fix the open issue and other people using it next week, but keen to get this right.